src/application/services/useLayout.ts

```diff
@@ -0,0 +1,23 @@
+import { shallowRef, ShallowRef } from 'vue';
+import router from '@/application/router';
+import DefaultLayout from '@/presentation/layouts/DefaultLayout.vue';
+
+const layouts = {
+  DefaultLayout,
+};
+
+/**
+ * useLayout hook is setting current Layout
+ *
+ * @returns { ShallowRef<string> } - Layout template was got by meta
+ */
+export default function (): ShallowRef<string> {
+  const layout = shallowRef('div');
+
+  router.afterEach((to) => {
+    layout.value = layouts[to.meta.layout] || 'div';
+  });
```

Comment:
I think, you should use the current route instead of looping over all routes

Reply:
Why is is loop? It's just once after rendering, because lProperty "layout" was accessed during render but is not defined on instance.